### DISCUSSION TITLE
O_MBAR Plot

### DISCUSSION THREAD

**Discussion by hl2500** [alchemistry/alchemlyb#294] [Q&A] [answered]
Hello, I have a simulation with lambdas that control bonded, coul and vdw interactions respectively. I was wondering if the lambda values in the O_MBAR plot correspond to the ones in the simulation? I'm asking because in the dhdl_TI plot, the lambdas do not correspond to ones in the simulation, with coul first, and then vdw and bonded (the sequence in the simulation is bonded, coul and then vdw). Are the lambda values in the O_MBAR plot the same as the ones in dhdl_TI plot? Thanks.

**Comment by orbeckst** [comment]
@xiki-tempula do you know anything about the ordering of lambdas in O_MBAR plot?

**Reply by xiki-tempula** [reply]
The ordering of lambdas in O_MBAR is just the order of the lambda states given in the MBAR.overlap_matirx. The dhdl_TI is more complicated as the original plotting function in the alchemical analysis doesn't respect the order of the lambda states and would intelligently sort them.
I don't have time now but I was thinking of looking deeply at the dhdl_TI plotting function.

**Comment by xiki-tempula** [comment, marked as answer]
@hl2500 Can I have your files? The MBAR plot is based on the ordering to the lambda states, which should be in the same order as the dF states plots. The dhdl_TI is more tricky and more difficult as it kind of did some guessing to combine certain stages together which can break in many cases. For example, if one has
```
coul = [0.0, 0.5, 1.0, 1.0, 1.0]
vdw = [ 0.0, 0.0, 0.0, 0.5, 1.0]
```
The dhdl_TI plot will combine
```
coul = [0.0, 0.5, 1.0]
vdw = [ 0.0, 0.0, 0.0]
```
to be the coul stage and
```
coul = [1.0, 1.0, 1.0]
vdw = [0.0, 0.5, 1.0]
```
to be the vdw stage.

However, if one has 
```
coul = [0.0, 0.5, 1.0, 1.0, 1.0]
vdw = [ 0.0, 0.0, 0.5, 0.5, 1.0]
```
Where there is an overlap between two stages, this script will break down and I'm not sure of what is the best way to fix it.

**Reply by hl2500** [reply]
Thank you for the information! Sorry, I couldn't find the dhdl files anymore. 

The TI plot generated by alchemical_analysis.py has an estimated curve. Is there a way to do this in alchemlyb? Thanks!

**Reply by xiki-tempula** [reply]
Sorry we don't do that in alchemlyb.
The  estimated curve in alchemical analysis comes from TI-cubic interpolation, which doesn't seems to be doing much better than pure TI. So we are not doing that here and thus no estimated curve.
